core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -674,7 +675,10 @@ bool core_mmu_divide_block(struct core_mmu_table_info *tbl_info,
 		return false;
 
 	entry = (uint64_t *)tbl_info->table + idx;
-	assert((*entry & DESC_ENTRY_TYPE_MASK) == BLOCK_DESC);
+	assert(!*entry || (*entry & DESC_ENTRY_TYPE_MASK) == BLOCK_DESC);
+
+	/* We need to flush TLBs only if there already was some mapping */
+	flush_tlb = *entry;
```

Comment:
If `*entry` contains some value, it can overflow the `bool`. Better normalize with `!!` before assingment.